src/expr.rs

```diff
@@ -1984,6 +1984,9 @@ pub(crate) fn rewrite_assign_rhs_with<S: Into<String>, R: Rewrite>(
 ) -> Option<String> {
     let lhs = lhs.into();
     let rhs = rewrite_assign_rhs_expr(context, &lhs, ex, shape, rhs_kind, rhs_tactics)?;
+    if context.config.version() == Version::Two && lhs.ends_with(" ") && rhs.starts_with(" ") {
+        return Some(lhs + &rhs.trim_start());
+    }
```

Comment:
Although this technically fixes the issue, it doesn't resolve the underlying problem. I believe the trailing space is properly added to the end of `"pub extern "`, but we shouldn't be adding a space before ` "C"`. I think the actual issue is occurring in `choose_rhs`. @calebcartwright Do you think this is the right approach?

Reply:
Okok. This approach also handles other similar cases. However, it there's a better approach I'm up to take it. I'll wait for confirmation before making any changes to `choose_rhs` 👍.